relay.go

```diff
@@ -0,0 +1,86 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// The returned RelayRequest struct can be marshalled and delivered to a service endpoint through an HTTP POST request.
+func BuildRelayRequest(
+	endpointSelector EndpointSelector,
+	requestBz []byte,
+) (*servicetypes.RelayRequest, error) {
+	if endpointSelector == nil {
+		return nil, errors.New("BuildRelayRequest: endpointSelector not specified")
+	}
+
+	sessionHeader, err := endpointSelector.SessionHeader()
+	if err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: could not get session header: %w", err)
+	}
+
+	if err := sessionHeader.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: error validating session header: %w", err)
+	}
+
+	supplierAddress, err := endpointSelector.SelectedSupplierAddress()
+	if err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: error getting a supplier: %w", err)
+	}
+
+	return &servicetypes.RelayRequest{
```

Comment:
I think we should sign the request before returning it. I see no use case where we would need an unsigned `RelayRequest` except for signing it :smile:

Reply:
`RelayRequest` is a simple struct, and building it is done here only as a helper/guide function.
Signing on the other hand involves multiple steps (and could potentially be applicable to other structs/payloads?).
This separation of building and signing of a request is done intentionally to allow building the `RelayRequest` in any other way, while still being able to use the signing functionality.
If you think changes are needed related to this please reopen this conversation.